### PR TITLE
SplitButton: Menu target should be the entire control instead

### PR DESCRIPTION
**Pull request by christiango**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes
A split button is a component with two buttons in it. The button on the left invokes an action and the button on the right invokes a menu. Currently, the target passed to the contexutal menu created when the menu launcher is clicked is just the button element on the left. This results in a few bugs:

1. Clicking the menu launcher when the menu is already open causes the menu to be opened again rather than dismissed. 
splitbuttonlauncher

2. The menu is positioned against the wrong element, this is most evident when using the rightTopEdge DirectionalHint on the menu
splitbuttonbug2

This PR updates the target of the menu launched by a split button to be the div that wraps the two buttons, instead of just the button on the left. 
.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -373,6 +374,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
         className={ css(disabled ? styles!.splitButtonContainerDisabled : styles!.splitButtonContainer) }
         tabIndex={ 0 }
         onKeyDown={ this.props.disabled ? undefined : this._onSplitButtonKeyDown }
+        ref={ div => this._splitButtonContainer = div }
```
[the remark this answers is not in the document]
Comment (christiango) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Yup, sounds reasonable. I'll update it.

[the remark this answers is not in the document]
Comment (cschleiden) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Or `BaseComponent`'s `resolveRef`?

[the remark this answers is not in the document]
Comment (christiango) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: This is getting resolved against a div not a component. I don't have a BaseComponent to work with here.

[the remark this answers is not in the document]
Comment (cschleiden) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: @christiango You could use

```TypeScript
ref={ this._resolveRef('_splitButtonContainer') }
```

Unfortunately (due to lack of a `nameof` operator in TS) it's not rename safe, but it's cached and should set your field

[the remark this answers is not in the document]
Comment (christiango) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Thanks for the tip! Updating.

**Pull request merged**